<!DOCTYPE html>
<html lang="en-us">

  <head>
  

    <meta charset="UTF-8">
    <title>CRESS-GLUE by giffordlabcvr: Paleovirology resources for CRESS DNA viruses</title>
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <link rel="stylesheet" type="text/css" href="../assets/stylesheets/normalize.css" media="screen">
    <link href='https://fonts.googleapis.com/css?family=Open+Sans:400,700' rel='stylesheet' type='text/css'>
    <link rel="stylesheet" type="text/css" href="../assets/stylesheets/stylesheet.css" media="screen">
    <link rel="stylesheet" type="text/css" href="../assets/stylesheets/github-light.css" media="screen">

  </head>

  <body>
  
  
    <section class="page-header">
    
      <h1 class="project-name">CRESS-GLUE</h1>
      
      <h2 class="project-tagline">Resources for comparative genomic analysis of CRESS DNA viruses.</h2>

      <a href="../../index.html" class="btn">Home</a>
      <a href="./viruses.html" class="btn">Viruses</a>
	  <a href="./tutorial.html"class="btn">Tutorial</a>
	  
      <a href="https://github.com/giffordlabcvr/CRESS-GLUE-EVE/zipball/master" class="btn">Download</a>
      <a href="https://github.com/giffordlabcvr/CRESS-GLUE-EVE/tree/master" class="btn">View on GitHub</a>
      
    </section>

    <section class="main-content">

	   <h3>
		 <a id="Background" class="anchor" href="#Background" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a><strong>Endogenous CRESS virus (ECV) data</strong>
	   </h3>
         <hr>

 		<p>
          These web pages provide a description of CRESS-GLUE-EVE, an extension to CRESS-GLUE 
          that extends CRESS-GLUE through the inclusion of CRESS DNA virus-derived endogenous 
          viral elements (EVEs). 

		</p>

		<p>
          Sequences derived from CRESS DNA viruses commonly occur in metazoan genomes. 
          These <b>endogenous cress virus (ECV)</b> provide unique information about 
          the evolution and biology of the <b><i>Cressdnaviricota</i></b>, being similar
          in some ways to '<b>viral fossils</b>'.
          ECV sequences have revealed a wealth of new information 
          about CRESS DNA virus evolution. 
          <b>CRESS-GLUE</b> represents and organises the fossil record of CRESS DNA viruses.
 		</p>
 


        <p><img src="../assets/images/ecv-header.jpg" alt="Species with endogenous CRESS DNA viruses" /></p>


<blockquote>
		<p>
		  Some of the species in which we identified EVEs that are derived from
		  <b>CRESS DNA viruses</b>, left to right:
		  icefish (<i>Chaenocephalus aceratus</i>),
		  wels catfish CRESS DNA virus (<i>Siluris glanis</i>),
		  black & white tegu (<i>Salvator merianae</i>),
		  Sunda flying lemur (<i>Galeopterus variegatus</i>).
		</p>
         
</blockquote>

	   <br>
		 


	   <h3>
		 <a id="Virology" class="anchor" href="#sequenceData" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a><strong>Relevance to virology</strong>
	   </h3>        
	   <hr>

	   <p>
          Endogenous viral sequences can inform our understanding of contemporary viruses
          in a wide variety of ways. Perhaps most importantly, EVEs allow calibration of
          the long-term evolutionary history of virus groups, which greatly influences 
          how we understand their biology.
 	   </p>

	   <p>
         Importantly, once time calibrations have been established, a far richer range of
         comparative genomic studies can be performed. By examining
         variation in the light of a known evolutionary history, these studies can provide 
         invaluable insights into the biological mechanisms through which viruses 
         replicate and spread.
	   </p>

	   <p>
          In some cases, studies of paleoviruses can go a step further -
          EVE sequences can be used to guide the reconstitution of functional nucleic acids
          and proteins via gene synthesis, so that their biological properties
          may be empirically investigated. This in turn, can greatly expand the scope of 
          comparative/theoretical approaches in paleovirology.
	   </p>

  
          
		 <br>

        <p><img src="../assets/images/ecv-midlevel.jpg" alt="Circovirus host species" /></p>



<blockquote>
		<p>
		  Some more of the species in which we identified EVEs derived from CRESS DNA viruses,
		  left to right: inshore hagfish (<i>Eptatretus burgeri</i>),
		  Ryukyu mouse (<i>Mus caroli</i>),
		  dust mites (<i>Tropilaelaps mercedesae</i>),
		  <i>Pseudomyrmex</i> ants.
		</p>
         
</blockquote>
		 
 
         
	   <br>
	   <h3>
		 <a id="Metagenomics" class="anchor" href="#sequenceData" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a><strong>Relevance to viral metagenomics</strong>
	   </h3>
       <hr>


	   <p>
          The ECV sequences in CRESS-GLUE can provide a useful resource for those
          interested in identifying and characterising CRESS DNA viruses in metagenomic
          datasets.
	   </p>


	   <p>
          Firstly, these sequences be used to exclude any potential 'false positive'
          hits (i.e. sequences that seem to represent new CRESS DNA viruses but in fact derive
          from genomic DNA).
	   </p>


	   <p>
          In addition, when new CRESS DNA virus species are identified, inclusion of EVEs
          in phylogenetic analyses can often provide useful information
          about their broader ecology and evolution, including (uniquely) their long-term
          evolution. 
          For example, see
		<a href="http://jvi.asm.org/content/92/16/e00145-18.full" target="_blank">Dennis et al. (2018)</a>, and 
		<a href="https://www.nature.com/articles/s41467-020-18474-w" target="_blank">Kinsella et al. (2020)</a>.
	   </p>
        
        
 
       
	   <br>
	   <h3>
		 <a id="Genomics" class="anchor" href="#sequenceData" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a><strong>Relevance to genomics</strong>
	   </h3>        
	   <hr>

	   <p>
          ECVs are not only useful genetic markers, several lines of evidence indicate
          they may have, or have had, functional roles as host alleles. The prevalence of 
          multicopy ECV lineages in some species suggests that germline incorporation of
          CRESS DNA virus sequences might have influenced the evolution of host genomes in
          important ways.
	   </p>
 
  
  
  
  
       
	   <br>
	   <h3>
		 <a id="dataProvenance" class="anchor" href="#dataProvenance" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a><strong>Where do the ECV data come from?</strong>
	   </h3>
       <hr>
 

	   <p>
		 EVE sequences were recovered from whole genome sequence (WGS) assemblies 
		 via <b>database-integrated genome screening (DIGS)</b> using the 
		  <a target="_blank" href="https://giffordlabcvr.github.io/DIGS-tool/"><b>DIGS tool</b></a>.
	   </p>
	   

	   <p>
		 All data pertaining to this screen are included in this repository, or in the 
		 associated "DIGS-for-EVEs" project.
	   </p>

   
	   <ul>

	   <li>
		 The complete list of <b>vertebrate</b> genomes screened can be found
         <a href="https://github.com/giffordlabcvr/CRESS-GLUE-EVE/blob/master/digs/vertebrate-genomes-screened.csv"><b>here</b></a>.
	   </li>

 
	   <li>
		 The complete list of <b>invertebrate</b> genomes screened can be found
         <a href="https://github.com/giffordlabcvr/CRESS-GLUE-EVE/blob/master/digs/invertebrate-genomes-screened.csv"><b>here</b></a>.
	   </li>
 
	   <li>
		 The set of <b>CRESS DNA virus</b> polypeptide sequences used as probes can be found
         <a href="https://github.com/giffordlabcvr/CRESS-GLUE-EVE/blob/master/digs/cress-probes.faa"><b>here</b></a>.
	   </li>
  
	   <li>
		 The final set of <b>CRESS DNA virus</b> and <b>ECV</b> polypeptide sequences used as
		 references can be found
         <a href="https://github.com/giffordlabcvr/CRESS-GLUE-EVE/blob/master/digs/cress_refs.faa"><b>here</b></a>.
	   </li>
   
	   <li>
		 Input parameters for screening using the
		 <a target="_blank" href="https://giffordlabcvr.github.io/DIGS-tool/"><b>DIGS tool</b></a>
		 can be found
         <a href="https://github.com/giffordlabcvr/CRESS-GLUE-EVE/blob/master/digs/eve_1_cress.ctl"><b>here</b></a>.
	   </li>
    
	   </ul>




	   <h3>
		 <a id="Nomenclature" class="anchor" href="#Nomenclature" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a><strong>Nomenclature for ECVs</strong>
	   </h3>
         <hr>

		<p>
          We have applied a <u>systematic approach to naming ECV</u>, following a convention 
          developed for <a target="_blank" href="https://retrovirology.biomedcentral.com/articles/10.1186/s12977-018-0442-1"><b>endogenous retroviruses (ERVs)</b></a>.
		  Each element was assigned a unique identifier (ID) constructed from a defined
		  set of components.
  		</p>
  		
  		
 
        <p><img src="../assets/images/ecv-nomenclature.png" alt="ECV Nomenclature" /></p>
  		

		<p>
          The <b>first component</b> is the classifier ‘ECV’ (endogenous CRESS virus). 
		</p>

		<p>
          The <b>second component</b> is a composite of two distinct subcomponents separated by a period:
          
          (i)  the name of ECV group;
          (ii) a numeric ID that uniquely identifies the insertion.
          The numeric ID is an integer that identifies a unique insertion locus that arose as a
          consequence of an initial germline infection. Thus, orthologous copies in different
          species are given the same number. 
		</p>

 
 		<p>
          Where an ECV sequence is thought to have been duplicated within the germline
          following it's initial incorporation (e.g. via segmental duplication or transposition)
          we have appended an additional 'duplicate id' to the numeric ID, separated by a period.
          Please note that we have not yet resolved the orthologous relationships among
          sets of eHBV sequences belonging to multicopy ECV lineages. We
          have therefore assigned unique duplicate IDs to each sequence within these lineages. 
		</p>

		
		<p>
          The <b>third component</b> of the ID defines the set of host species in which
          the ortholog occurs, or did occur prior to being deleted.
 		</p>


         
		 <br>

	   <h3>
		 <a id="ECV reference sequences and data" class="anchor" href="#Ref_ecv_data" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a><strong>ECV reference sequences and data</strong>
	   </h3>
         <hr>

		<p>
           We constructed reference sequences for using alignments of ECV
           sequences derived from the same initial germline colonisation event - i.e.
           orthologous elements in distinct species, and paralogous 
           elements that have arisen via intragenomic duplication of ECV sequences.
  		</p>


		<p>
           Reference sequence data in tabular format can be found at the following links: 

		  <ul>
			<li> ECVs derived from genus <a href="https://github.com/giffordlabcvr/CRESS-GLUE-EVE/blob/master/tabular/eve/circoviridae/ecv-circo-refseqs-side-data.tsv"><strong><i>Circovirus</i></strong></a></li>
			<li> ECVs derived from genus <a href="https://github.com/giffordlabcvr/CRESS-GLUE-EVE/blob/master/tabular/eve/circoviridae/ecv-cyclo-refseqs-side-data.tsv"><strong><i>Cyclovirus</i></strong></a></li>
			<li> ECVs derived from <a href="https://github.com/giffordlabcvr/CRESS-GLUE-EVE/blob/master/tabular/eve/circoviridae/ecv-cress-refseqs-side-data.tsv"><strong>unclassified CRESS DNA viruses</strong></a></li>
		  </ul>
           
  		</p>
  		
		<br>

		<p>
           <strong>Nucleotide level data:</strong>

         
		<br>

		   <table>
	   
			   <thead>
				   <tr>
					   <th>Taxonomic group</th>
					   <th>Tabular data</th>
					   <th>Full-length ECV nucleotides</th>
					   <th>Full-length ECV</th>
				   </tr>
			   </thead>
	   
			   <tbody>

				   <tr>
					   <td><i>Genus Circovirus</i></td>
					   <td><a href="https://github.com/giffordlabcvr/CRESS-GLUE-EVE/blob/master/tabular/eve/circoviridae/ecv-circo-refseqs-side-data.tsv"><i>2020-07-15</i></a></td>
					   <td><a href="../alignments/internal/circoviridae/circovirus-genome+eves.fna">FASTA MSA</a></td>
					   <td><a href="https://github.com/giffordlabcvr/CRESS-GLUE-EVE/tree/master/sources/eve/circoviridae/fasta-refseqs-circo-ecv">Individual FASTA</a></td>
				   </tr>

				   <tr>
					   <td><i>Genus Cyclovirus</i></td>
					   <td><a href="https://github.com/giffordlabcvr/CRESS-GLUE-EVE/blob/master/tabular/eve/circoviridae/ecv-cyclo-refseqs-side-data.tsv"><i>2020-07-15</i></a></td>
					   <td><a href="../alignments/internal/circoviridae/cyclovirus-rep+eves.fna">FASTA MSA</a></td>
					   <td><a href="https://github.com/giffordlabcvr/CRESS-GLUE-EVE/tree/master/sources/eve/circoviridae/fasta-refseqs-cyclo-ecv">Individual FASTA</a></td>
				   </tr>

				   <tr>
					   <td><i>Unclassified CRESS</i></td>
					   <td><a href="https://github.com/giffordlabcvr/CRESS-GLUE-EVE/blob/master/tabular/eve/circoviridae/ecv-cress-refseqs-side-data.tsv"><i>2020-07-15</i></a></td>
					   <td>-</td>
					   <td><a href="https://github.com/giffordlabcvr/CRESS-GLUE-EVE/tree/master/sources/eve/circoviridae/fasta-digs-unc-cress-ecv">Individual FASTA</a></td>
				   </tr>

			   </tbody>
		   
		   </table>

		 <br>
		 <br>


  		</p>

		<p>
           <strong>Protein level data:</strong>
  		</p>
		
		<p>
		<table>
		
			<thead>
				<tr>
					<th>Taxonomic group</th>
					<th>Rep protein</th>
					<th>Rep protein</th>
					<th>Cap protein</th>
					<th>Cap protein</th>
				</tr>
			</thead>
		
			<tbody>

				<tr>
					<td><i>Genus Circovirus</i></td>
					<td><a href="https://github.com/giffordlabcvr/CRESS-GLUE-EVE/blob/master/export/ecv-circo-rep.polypeptide.aln.faa">FASTA MSA</a></td>
					<td><a href="https://github.com/giffordlabcvr/CRESS-GLUE-EVE/tree/master/export/ecv-circo-rep-polypeptide-fasta">Individual FASTA</a></td>
					<td><a href="https://github.com/giffordlabcvr/CRESS-GLUE-EVE/blob/master/export/ecv-circo-cap.polypeptide.aln.faa">FASTA MSA</a></td>
					<td><a href="https://github.com/giffordlabcvr/CRESS-GLUE-EVE/tree/master/export/ecv-circo-cap-polypeptide-fasta">Individual FASTA</a></td>
				</tr>


				<tr>
					<td><i>Genus Cyclovirus</i></td>
					<td><a href="https://github.com/giffordlabcvr/CRESS-GLUE-EVE/blob/master/export/ecv-circo-rep.polypeptide.aln.faa">FASTA MSA</a></td>
					<td><a href="https://github.com/giffordlabcvr/CRESS-GLUE-EVE/tree/master/export/ecv-circo-rep-polypeptide-fasta">Individual FASTA</a></td>
					<td><a href="https://github.com/giffordlabcvr/CRESS-GLUE-EVE/blob/master/export/ecv-circo-cap.polypeptide.aln.faa">FASTA MSA</a></td>
					<td><a href="https://github.com/giffordlabcvr/CRESS-GLUE-EVE/tree/master/export/ecv-circo-cap-polypeptide-fasta">Individual FASTA</a></td>
				</tr>


			</tbody>
			
		</table>

  		</p>
   

         
		 <br>


	   <h3>
		 <a id="Multiple sequence alignments" class="anchor" href="#MultipleSequenceAlignments" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a><strong>Multiple sequence alignments</strong>
	   </h3>
         <hr>

		<p>

          Multiple sequence alignment constructed in this study are linked together using
          GLUE's 
          ‘<a href="./images/alignmentTree.jpg">alignment tree</a>’
          data structure. Alignments in the project include:

		  <ol>
			<li> A single ‘<a href="https://github.com/giffordlabcvr/CRESS-GLUE-EVE/blob/master/alignments/root/cress-root+eve.fna"><strong>root</strong></a>’
			alignment constructed to represent proposed
			homologies between representative members of major CRESS virus lineages
			(including extinct lineages represented only by ECVs).</li>
			<li> ‘<a href="https://github.com/giffordlabcvr/CRESS-GLUE-EVE/tree/master/alignments/internal"><strong>Genus-level</strong></a>’
			alignments for each CRESS DNA virus family.
			These alignments have been constructed to represent homologies between the genomes of
			representative members of specific CRESS virus genera and ECV reference sequences.
			<li> ‘<a href="https://github.com/giffordlabcvr/CRESS-GLUE-EVE/tree/master/alignments/tip"><strong>Tip</strong></a>’
			alignments in which all taxa are derived from a single ECV lineage.</li>
		  </ol>


 		</p>

         
		<br>



		<h3>
		  <a id="Phylogenetic trees" class="anchor" href="#PhylogeneticTrees" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a><strong>Phylogenetic trees</strong>
		</h3>
        <hr>

		<p>

          We used GLUE to implement an automated process for deriving midpoint rooted,
          annotated trees from the alignments included in our project.
 		</p>


		<p>
          Trees were constructed at distinct taxonomic levels:
        
		  <ol>
			<li> Recursively populated <a href="https://github.com/giffordlabcvr/CRESS-GLUE-EVE/tree/master/trees/phylum-root"><strong>root</strong></a> phylogeny (Rep)</li>
			<li> <a href="https://github.com/giffordlabcvr/CRESS-GLUE-EVE/tree/master/trees/genus-level-with-eves/"><strong>Genus-level</strong></a> phylogenies</li>
			<li> <a href="https://github.com/giffordlabcvr/CRESS-GLUE-EVE/tree/master/trees/eve-orthologs/"><strong>ECV lineage-level</strong></a> phylogenies</li>
		  </ol>

 		</p>


        <br>

	   <h3>
		 <a id="Raw ECV sequences and data" class="anchor" href="#Raw_ecv_data" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a><strong>Raw ECV sequences and data</strong>
	   </h3>
         <hr>

		 <p>
		   These are the raw data generated by
		   <a target="_blank" href="https://giffordlabcvr.github.io/DIGS-tool/">database-integrated genome screening (DIGS)</a>.
		   The tabular files contain information about the genomic location of each EVE. 
		   EVEs were classified by comparison to a polypeptide sequence reference library
		   designed to represent the known diversity of CRESS viruses - this includes extinct 
		   lineages represented only by endogenous viral elements (EVEs).
		 </p>

		 <p>
		   These data were obtained via
		   <a target="_blank" href="https://giffordlabcvr.github.io/DIGS-tool/">DIGS</a>
		   performed in vertebrate genome assemblies downloaded from 
		   <a target="_blank" href="https://www.ncbi.nlm.nih.gov/genome/browse ">NCBI genomes</a>
		   (2020-07-15).
		 </p>

		 <table>
		
			 <thead>
				 <tr>
					 <th>Taxonomic Group</th>
					 <th>ECV locus data in tabular format</th>
					 <th>Individual FASTA files</th>
				 </tr>
			 </thead>
		
			 <tbody>

				 <tr>
					 <td>Genus <i>Circovirus</i></a></td>
					 <td><a href="https://github.com/giffordlabcvr/CRESS-GLUE-EVE/blob/master/tabular/eve/circoviridae/ecv-circo-side-data.tsv">2020-07-15</a></td>
					 <td><a href="https://github.com/giffordlabcvr/CRESS-GLUE-EVE/tree/master/sources/fasta-digs-circo-ecv">2020-07-15</a></td>
				 </tr>


				 <tr>
					 <td>Genus <i>Cyclovirus</i></a></td>
					 <td><a href="https://github.com/giffordlabcvr/CRESS-GLUE-EVE/blob/master/tabular/eve/circoviridae/ecv-cyclo-side-data.tsv">2020-07-15</a></td>
					 <td><a href="https://github.com/giffordlabcvr/CRESS-GLUE-EVE/tree/master/sources/fasta-digs-cyclo-ecv">2020-07-15</a></td>
				 </tr>

				 <tr>
					 <td>Unclassified <i>CRESS</i></a></td>
					 <td><a href="https://github.com/giffordlabcvr/CRESS-GLUE-EVE/blob/master/tabular/eve/circoviridae/ecv-cress-side-data.tsv">2020-07-15</a></td>
					 <td> <a href="https://github.com/giffordlabcvr/CRESS-GLUE-EVE/tree/master/sources/fasta-digs-cress-ecv">2020-07-15</a></td>
				 </tr>

			 </tbody>
			
		 </table>
         
		<br>

        
       <br>
	   <h3>
		 <a id="schemaExtensions" class="anchor" href="#sequenceData" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a><strong>Paleovirus-specific schema extensions</strong>
	   </h3>
       <hr>



		 <p>
		   The paleovirus component of CRESS-GLUE extends GLUE's
		   <a target="_blank" href="http://glue-tools.cvr.gla.ac.uk/#/coreSchema"><b>core schema</b></a>
		   to allow the capture of EVE-specific data.
		   These schema extensions are defined in 
           <a href="https://github.com/giffordlabcvr/CRESS-GLUE-EVE/blob/master/glue/core/cressSchemaExtensionsPaleovirus.glue"><b>this file</b></a>
		   and comprise two additional tables: '<b>locus_data</b>' and '<b>refcon_data'</b>.
		   Both tables are linked to the main '<b>sequence</b>' table via the 'sequenceID' field.
  		</p>


 		<p>
		   The '<b>locus_data</b>' table contains information pertaining to individual
		   EVE sequences: e.g. species in which they occur, genome assembly version, 
		   genomic location(i.e. scaffold, location coordinates, and orientation).
		</p>

 		<p>
		   The '<b>refcon_data</b>' table contains information pertaining to our
		   ECV reference sequences, which we have constructed in an effort to
		   reconstruct, as closely as possible, the sequences of the progenitor viruses
		   that gave rise to EVEs. 
		</p>



        
       <br>
	   <h3>
		 <a id="Publications" class="anchor" href="#Publications" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a><strong>Related Publications</strong>
	   </h3>
       <hr>

        Kinsella CM, Bart A, Deijs M, Broekhuizen P, Kaczorowska J, Jebbink MF, van Gool T,
        Cotten M, and L van der Hoek.
		<b>(2020)</b>
		<br>
        Entamoeba and Giardia parasites implicated as hosts of CRESS viruses.
		<br>
		<i>Nat Commun.</i> Sep 15;11(1):4620. doi: 10.1038/s41467-020-18474-w. 
		<a href="https://www.nature.com/articles/s41467-020-18474-w" target="_blank">[view]</a> 
		<br>
		<br>

		Dennis TPW, de Souza WM, Marsile-Medun S, Singer JB, Wilson SJ, and RJ Gifford 
		<b>(2019)</b>
		<br>
		The evolution, distribution and diversity of endogenous circoviral elements in vertebrate genomes.
		<br>
		<i>Virus Research</i>
		<a href="https://www.sciencedirect.com/science/article/pii/S0168170217309048" target="_blank">[view]</a> 
		<br>
		<br>

		Dennis TPW, Flynn PJ, de Souza WM, Singer JB, Moreau CS, Wilson SJ, and RJ Gifford 
		<b>(2018)</b>
		<br>
		Insights into CRESS DNA virus host range from the genomic fossil record.
		<br>
		<i>Journal of Virology</i>
		<a href="http://jvi.asm.org/content/92/16/e00145-18.full" target="_blank">[view]</a> 
		<br>
		<br>

		Singer JB, Thomson EC, McLauchlan J, Hughes J, and RJ Gifford
		<b>(2018)</b>
		<br>
		GLUE: A flexible software system for virus sequence data.
		<br>
		<i>BMC Bioinformatics</i>
		<a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-018-2459-9" target="_blank">[view]</a> 
		<br>
		<br>

		Zhu H, Dennis T, Hughes J, and RJ Gifford
		<b>(2018)</b>
		<br>
		Database-integrated genome screening (DIGS): exploring genomes heuristically using sequence similarity search tools and a relational database.
		<a href="https://www.biorxiv.org/content/early/2018/04/25/246835" target="_blank">[preprint]</a> 
		<br>
		<br>

		Gifford RJ, Blomberg B, Coffin JM, Fan H, Heidmann T, Mayer J, Stoye J, Tristem M, and WE Johnson
		<b>(2018)</b>
		<br>
		Nomenclature for endogenous retrovirus (ERV) loci.
		<br>
		<i>Retrovirology</i>
		<a href="https://retrovirology.biomedcentral.com/articles/10.1186/s12977-018-0442-1" target="_blank">[view]</a> 
		<br>
		<br>

		Katzourakis A. and RJ. Gifford
		<b>(2010)</b>
		<br>
		Endogenous viral elements in animal genomes.
		<br>
		<i>PLoS Genetics</i>
		<a href="http://www.plosgenetics.org/article/info:doi/10.1371/journal.pgen.1001191" target="_blank">[view]</a>

 		</p>


	 <footer class="site-footer">
	   <span class="site-footer-owner"><a href="https://github.com/giffordlabcvr/CRESS-GLUE-EVE/">CRESS-GLUE</a> is maintained by <a href="https://github.com/giffordlabcvr">giffordlabcvr</a>.</span>

	   <span class="site-footer-credits">This page was generated by <a href="https://pages.github.com">GitHub Pages</a> using the <a href="https://github.com/jasonlong/cayman-theme">Cayman theme</a> by <a href="https://twitter.com/jasonlong">Jason Long</a>.</span>
	 </footer>

    </section>


  </body>
  
</html>



