<div class="row tim-row">


<div class="col-md-8 col-md-offset-2">


### A sequence data resource for circoviruses

Our circovirus study is part of a larger investigation into the diversity of endogenous
viral elements (EVEs) – virus-derived DNA sequences that are found in the genomes of
animals, plants, fungi, and other eukaryotic species. EVE sequences are highly informative
about the long-term interactions of viruses and hosts. They are also interesting because
they have influenced the evolution of mammalian genomes in myriad ways.


<div class="picture2 right2" style="width:300px;">
  <a href="https://twitter.com/DIGS_for_EVEs" target="_blank">
  <img src="../../assets/paper_img/blog/circovirus/digs-for-eves.jpg" width="300" height="203" alt="digs-for-eves project"/>
  </a>
  The progress of the DIGS-for-EVEs project is being tracked in a twitter stream
</div>


We developed a genome-mining approach called
[database-integrated genome screening](https://giffordlabcvr.github.io/DIGS-tool/)
(DIGS). We’re using DIGS to screen eukaryotic genome assemblies for EVEs, as part of a
project called 
[DIGS for EVEs](http://giffordlabcvr.github.io/DIGS-for-EVEs/)
We are cataloguing EVEs like this because (i) they are important, informative
sequences, and (ii) due to their erratic distribution and/or advanced state of decay,
they are not yet reliably annotated.

In fact, we’re aiming to go a little further than simply annotating EVE sequences. Our
goal is to make use of
[GLUE](http://tools.glue.cvr.ac.uk/),
a bioinformatics software environment developed in our lab, to organise EVE data at a
higher level for specific virus groups, so that we create sequence-based resources for
the virus families that have generated EVE sequences. We picked the *Circoviridae*
family first, mainly because these viruses have small, simple genomes, so should hopefully
be among the more straightforward groups to deal with.

We collected together all the data used and generated in our study in a GLUE project
contained within the DIGS-for-EVEs repository. The circovirus-glue project contains 19
exogenous circovirus reference sequences, and 204 *Circovirus*-derived CVe - the number is
higher than in our paper because more vertebrate genomes have been sequenced and published
since we first performed this work, and we’ve found additional CVe in these recently
published genomes.

In the GLUE project, we've captured auxiliary data asssociated with both exogenous
reference sequences and CVe. Viral reference sequences are associated with information
such as isolate/strain ID, country of sampling, and taxonomy, while CVe are associated
with information about their genomic location and orientation within specific chromosome
assemblies or genomic contigs.

The circovirus project we have created using GLUE can serve in the first instance as a
source of reference information for a range of circovirus-related projects.
The project,
[which is openly available here](https://github.com/giffordlabcvr/DIGS-for-EVEs)
does not yet have a web interface - in other words, to make use of it you will need to
download the project and work locally (i.e. on your laptop or shared server).

Obviously, GLUE software can be used to interact with the circovirus GLUE project, but
also the project data can be utilised directly. For example, if someone is only interested
in obtaining the sequences, alignments and auxiliary data collated in our circovirus
project, these are easy to locate and retrieve.

However, working within the GLUE framework allows the project to be much more than than 
just a data repository, instead becoming a template for the development of a wide
range of genomics-oriented projects focussing on different aspects of circovirus biology,
and containing algorithms in addition to data.



        <div class="picture2 left2" style="width:200px;">
          <a href="http://www.healthmap.org/site/diseasedaily/article/circovirus-spreading-dogs-101613" target="_blank">
          <img src="../../assets/paper_img/blog/circovirus/dog.jpg" width="200" height="200" alt="dog versus headwind"/>
          </a>
           An ill wind? Some reports have linked canine circovirus infections with 
           outbreaks of severe disease in dogs.
        </div>



For example, some reports have suggested the possible emergence of a
[pathogenic circovirus](http://www.nbc12.com/story/23950879/special-report-deadly-dog-virus")
in dogs, but
[uncertainty remains](http://news.vin.com/VINNews.aspx?articleId=28696)
as to whether circovirus infection is responsible for disease in these cases.

A GLUE project could be created to focus specifically on using genomic data to
investigate the epidemiology of canine circoviruses as part of effort to
investigate this issue.

'Reuse' is a core theme of the GLUE framework. To create the project described above, data
components of the circovirus project constructed here could be combined with the
algorithmic elements of other GLUE projects focussed on genomic surveillance of
viral pathogens (e.g. our rabies project, to be discussed in a future blog).
This woud allow a framework for the proposed analysis to be created very
rapidly. Moreover, using GLUE's 
[extensible schema](http://tools.glue.cvr.ac.uk/#/schemaExtensions),
a resource could be created that captures any data specific to the etiology of
canine circovirus infections (e.g. information about 
[co-infection with canine parvovirus](https://www.sciencedirect.com/science/article/pii/S0042682216000106)

To give an alternative example, a project could be created to focus on analysis of RNA
secondary structures within the non-coding regions of circovirus genomes (see section above).
This project could include project code, scripts and modules that facilitate the export
and import of data from and to programs for RNA secondary structure prediction programs.
GLUE has been designed explicitly to facilitate these kinds of integrations between
bioinformatics programs and workflows.

</div>

</div>
