<!DOCTYPE html>
<html lang="en-us">

  <head>

    <meta charset="UTF-8">
   <title>CRESS-GLUE by giffordlabcvr: Comparative analysis of CRESS DNA viruses</title>
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <link rel="stylesheet" type="text/css" href="stylesheets/normalize.css" media="screen">
    <link href='https://fonts.googleapis.com/css?family=Open+Sans:400,700' rel='stylesheet' type='text/css'>
    <link rel="stylesheet" type="text/css" href="stylesheets/stylesheet.css" media="screen">
    <link rel="stylesheet" type="text/css" href="stylesheets/github-light.css" media="screen">

  </head>

  <body>

    <section class="page-header">
      <h1 class="project-name">CRESS-GLUE</h1>
      
      <h2 class="project-tagline">GLUE-based resources for comparative genomic analysis of CRESS DNA viruses.</h2>

      <a href="./website/project.html" class="btn">Viruses</a>
      <a href="./website/paleovirology.html" class="btn">Paleoviruses</a>
      <a href="https://github.com/giffordlabcvr/CRESS-GLUE/zipball/master" class="btn">Download .zip</a>
      <a href="https://github.com/giffordlabcvr/CRESS-GLUE/tree/master" class="btn">View on GitHub</a>
    </section>

    <section class="main-content">

	   <h3>
		 <a id="overview" class="anchor" href="#overview" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a><strong>Overview</strong>
	   </h3>

		 <p>
		   This is <b>CRESS-GLUE</b>, a GLUE
		   project designed to support comparative genomic and evolutionary analysis of
		   circular Rep-encoding single-stranded DNA
		   (<a target="_blank" href="https://www.sciencedirect.com/science/article/pii/S0065352718300575"><b>CRESS DNA</b></a>)
		   viruses (phylum 
		   <a target="_blank" href="https://jvi.asm.org/content/94/12/e00582-20"><b><i>Cressdnaviricota</i></b>)</a>. 
		 </p>


		 <p>
		   The
		   <b>CRESS-GLUE</b>
		   project contains a richly annotated sequence dataset comprised of both 
		   <a href="./website/project.html"><b>CRESS DNA viruses</b></a>
		   and
		   <a href="./website/paleovirology.html"><b>endogenous viral elements (EVEs)</b></a>
		   derived from these viruses.
		   
		 </p>


		 <p>
		   It has been designed to facilitate any form of comparative genomic analysis
		   involving these viruses.
		 </p>


		 <br>
        
         <p><img src="./website/images/cress-glue-footer.jpg" alt="CRESS DNA virus phylogeny" /></p>
         
         
		   <br>

		 <p>
		 
		   The figure above shows an
		   unrooted maximum likelihood phylogenetic tree of		   
		   <a target="_blank" href="https://www.uniprot.org/uniprot/Q8BB16"><b>Rep</b></a>
		   proteins from CRESS DNA viruses, taken from
		   <a target="_blank" href="https://jvi.asm.org/content/94/12/e00582-20"><b>Krupovic <i>et al.</i></b></a>
		 </p>
         <hr>

	   <h3>
	   <a id="glueProject" class="anchor" href="#glueProject" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a><strong>Building this GLUE project</strong>
	   </h3>

      	   
	   <p>
	
         On computers with <a target="_blank" href="http://glue-tools.cvr.gla.ac.uk/#/installation">GLUE installed</a>,
         the <b>CRESS-GLUE</b> project can be instantiated by navigating to the project folder,
         initiating GLUE, and issuing the following command in the GLUE shell:
         
  <pre>
  <code>Mode path: /
  GLUE> run file cressProject.glue</code></pre>
		 
	   </p>



	   <h3>
	   <a id="WhatGlueProject" class="anchor" href="#glueProject" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a><strong>What is a GLUE project?</strong>
	   </h3>


	   <p>
		 <a target="_blank" href="http://glue-tools.cvr.gla.ac.uk/"><b>GLUE</b></a> is an open, integrated
		 software toolkit that provides functionality for storage and interpretation of
		 sequence data. 
	   </p>

	   <p>
		 <b>GLUE</b> supports the development of <b>“projects”</b> containing the data items
		 required for comparative genomic analysis 
		 (e.g.  sequences, multiple sequence alignments, genome feature annotations,
		 and other sequence-associated data). 
	   </p>

	   <p>
		 Projects are loaded into the <b>GLUE "engine"</b>, creating a relational database
		 that represents the semantic relationships between data items.
		 This provides a robust foundation for the implementation of <b>systematic
		 comparative analyses</b> and the development of <b>sequence-based resources</b>. 
	   </p>

	   <p>		 
		  The core schema of this database can be extended to accommodate
		  the idiosyncrasies of different projects, and
		  GLUE provides a scripting layer (based on JavaScript) for developing custom
		  analysis tools. 
	   </p>


	   <p>
		  Hosting of GLUE projects in an online version control system (e.g. GitHub) provides
		  a mechanism for their stable, collaborative development.
	   </p>
	   
	   
	   
	   
	   <h3>
	   <a id="WhyGlueProject" class="anchor" href="#glueProject" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a><strong>What does building the CRESS-GLUE project offer?</strong>
	   </h3>
 
 
	   <p>
		 
		  CRESS-GLUE offers a number of advantages for performing comparative sequence
		  analysis of CRESS DNA viruses:

		 <ol>
		   <li> <strong>Standardisation of sequence orientation</strong>. Like many 
		   viruses with circular genomes, the orientation of circovirus sequences 
		   is not standardised in GenBank. CRESS-GLUE addresses this issue.
		   </li>
		   <li> <strong>Standardisation of the genomic co-ordinate space</strong>. GLUE
		   projects allow all sequences to utilise the coordinate space of a chosen 
		   reference sequence. Contingencies associated with insertions and deletions
		   (indels) are handled in a systematic way.
		   </li>
		   <li> <strong>Predefined, fully annotated reference sequences</strong>:
		   This project includes fully-annotated reference sequences for major lineages
		   within the CRESS DNA assemblage.
		   </li>
		   <li> <strong>Alignment trees</strong>:
		   GLUE allows linking of alignments constructed at distinct taxonomic levels
		   via an "alignment tree" data structure. In the alignment tree, each alignment
		   is constrained to a standard reference sequence, thus all multiple sequence
		   alignments are linked to one another via a standardised coordinate system.

		   </li>
		 </ol>

	   </p>



	   <h3>
		 <a id="Publications" class="anchor" href="#Publications" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a><strong>Related Publications</strong>
	   </h3>

         Krupovic M, Varsani A, Kazlauskas D, Breitbart M, Delwart E, Rosario K, Yutin N,
         Wolf YI, Harrach B, Zerbini FM, Dolja VV, Kuhn JH, and EVE Koonin.
		<b>(2020)</b>
		<br>
		Cressdnaviricota: a Virus Phylum Unifying Seven Families of Rep-Encoding Viruses with Single-Stranded, Circular DNA Genomes.
		<br>
		<i>J. Virol.</i>. Jun 1;94(12):e00582-20. doi: 10.1128/JVI.00582-20.
		<a href="https://jvi.asm.org/content/94/12/e00582-20" target="_blank">[view]</a> 
		<br>
		<br>



        Tisza MJ, Pastrana DV, Welch NL, Stewart B, Peretti A, Starrett GJ, Pang YS,
        Krishnamurthy SR, Pesavento PA, McDermott DH, Murphy PM, Whited JL, Miller B,
        Brenchley J, Rosshart SP, Rehermann B, Doorbar J, Ta'ala BA, Pletnikova O,
        Troncoso JC, Resnick SM, Bolduc B, Sullivan MB, Varsani A, Segall AM, Buck CB.

        
		<b>(2018)</b>
		<br>
		Discovery of several thousand highly diverse circular DNA viruses.
		<br>
		<i>Elife</i>. Feb 4;9:e51971. doi: 10.7554/eLife.51971.
		<a href="https://elifesciences.org/articles/51971" target="_blank">[view]</a> 
		<br>
		<br>



		Singer JB, Thomson EC, McLauchlan J, Hughes J, and RJ Gifford
		<b>(2018)</b>
		<br>
		GLUE: A flexible software system for virus sequence data.
		<br>
		<i>BMC Bioinformatics</i>
		<a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-018-2459-9" target="_blank">[view]</a> 
		<br>
		<br>


 		</p>



	   <h3>
		 <a id="contributors" class="anchor" href="#contributors" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a><strong>Contributors</strong>
	   </h3>
	   
	   <p>
		  Robert J. Gifford (<a href="mailto:robert.gifford@glasgow.ac.uk">robert.gifford@glasgow.ac.uk</a>)	 
	   </p>
	   <p>
          Tristan Dennis (<a href="mailto:tristan.dennis@glasgow.ac.uk>">tristan.dennis@glasgow.ac.uk</a>)
	   </p>

	   <h3>
		 <a id="license" class="anchor" href="#license" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a><strong>License</strong>
	   </h3>

	   <p>
		 This project is licensed under the <a href="https://www.gnu.org/licenses/agpl-3.0.en.html">GNU Affero General Public License v. 3.0</a>.
	   </p>

		<br>
		<br>


	 <footer class="site-footer">
	   <span class="site-footer-owner"><a href="https://github.com/giffordlabcvr/CRESS-GLUE">CRESS-GLUE</a> is maintained by <a href="https://github.com/giffordlabcvr">giffordlabcvr</a>.</span>

	   <span class="site-footer-credits">This page was generated by <a href="https://pages.github.com">GitHub Pages</a> using the <a href="https://github.com/jasonlong/cayman-theme">Cayman theme</a> by <a href="https://twitter.com/jasonlong">Jason Long</a>.</span>
	 </footer>

    </section>


  </body>
  
</html>
