<!DOCTYPE html>
<html lang="en-us">

  <head>
  
    <!-- Global site tag (gtag.js) - Google Analytics -->
	<script async src="https://www.googletagmanager.com/gtag/js?id=UA-127420264-1"></script>
	<script>
	  window.dataLayer = window.dataLayer || [];
	  function gtag(){dataLayer.push(arguments);}
	  gtag('js', new Date());
	  gtag('config', 'UA-127420264-1');
	</script>


    <meta charset="UTF-8">
    <title>CRESS-GLUE by giffordlabcvr</title>
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <link rel="stylesheet" type="text/css" href="../stylesheets/normalize.css" media="screen">
    <link href='https://fonts.googleapis.com/css?family=Open+Sans:400,700' rel='stylesheet' type='text/css'>
    <link rel="stylesheet" type="text/css" href="../stylesheets/stylesheet.css" media="screen">
    <link rel="stylesheet" type="text/css" href="../stylesheets/github-light.css" media="screen">

  </head>

  <body>
    <section class="page-header">
      <h1 class="project-name">CRESS-GLUE</h1>
      <h2 class="project-tagline">Paleovirology resources for CRESS DNA viruses.</h2>
      <a href="../index.html" class="btn">Home</a>
      <a href="./project.html" class="btn">Viruses</a>
      <a href="https://github.com/giffordlabcvr/CRESS-GLUE/zipball/master" class="btn">Download .zip</a>
      <a href="https://github.com/giffordlabcvr/CRESS-GLUE/tree/master" class="btn">View on GitHub</a>
    </section>

    <section class="main-content">


	   <h3>
		 <a id="tutorialOverview" class="anchor" href="#tutorialOverview" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a><strong>How to use CRESS-GLUE - the paleovirus fossil record</strong>
	   </h3>


	   <p>
         
         This <b>tutorial</b> comprises the following steps:

		 <ol>
		   <li> <b>Exporting</b> EVE sequences and data from CRESS-GLUE.</li>
		   <li> <b>Extracting</b> isolate data from GenBank files.</li>
		   <li> <b>Filtering</b> sequence data to obtain a valid analysis set.</li>
		   <li> <b>Defining virus reference sequences</b> and associating them with annotations.</li>
		   <li> <b>Constructing an alignment</b> using a codon-aware method.</li>
		   <li> <b>Mapping feature coverage</b> of sequence members within an alignment.</li>
		   <li> Reconstructing annotated, midpoint-rooted <b>phylogenetic trees</b>.</li>
		   <li> Using GLUE projects as a basis for <b>collaborative research and resource development</b>.</li>
		 </ol>


 	   </p>
 	   


	   <h3>
		 <a id="Background" class="anchor" href="#Background" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a><strong>Background - the 'fossil record' of CRESS DNA viruses</strong>
	   </h3>

        <p><img src="./images/parrots-header.png" alt="Parrots" /></p>

		<p>

		  This page provides a demonstration of CRESS-GLUE being used to 
		  (i) organise the fossil record and; (ii) expand the fossil record to 
		  accomodate new data. 

		</p>




	   <h3>
		 <a id="PaleovirologyUses" class="anchor" href="#PaleovirologyUses" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a><strong>Organising the fossil record</strong>
	   </h3>


		<p>

		  We used DIGS to identify the complete repertoire of ECV sequences in
		  published vertebrate genome assemblies. 
		  Through comparative analysis of these sequences
		  we could establish a picture yet of how viruses in the genus Circovirus
		  have evolved and interacted with their hosts over the course of their evolution. 

		</p>


		<p>

          ECV have previously been identified in the genome of the short-tailed opossum
          (Monodelphis domestica), an American marsupial. In phylogenies based on rep,
          this sequence groups together with the porcine circoviruses, canine circovirus,
          and an ECV identified in the Steppe mouse (Mus caroli). We identified the first examples of ECV
          from the genomes of Australian marsupial species: the Tasmanian devil
          (Sarcophilus harrisii) and the koala (Phascolarctos cinereus). 
          Both these sequences derived from circovirus rep genes, and grouped together 
          in phylogenetic trees. However, their placement relative to other 
          taxa was not supported with confidence, reflecting their short and degraded
          nature. Several other short and degraded matches to Rep probes were identified 
          in other mammalian species. These sequences were relatively distantly related 
          to one another and to contemporary circoviruses.

		</p>

	   <h3>
		 <a id="How to use these resources" class="anchor" href="#PaleovirologyUses" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a><strong>Expanding the fossil record</strong>
	   </h3>

		<p>

         As you might imagine, there will be more EVEs in the published data WGS now - ones that we havenâ€™t described. But importantly, it is now possible to pull these into the existing dataset quite painlessly. Works like this:

Scenario 1. Update to WGS data - screen with DIGS - classify new

		</p>







	 <footer class="site-footer">
	   <span class="site-footer-owner"><a href="https://github.com/giffordlabcvr/CRESS-GLUE/">CRESS-GLUE</a> is maintained by <a href="https://github.com/giffordlabcvr">giffordlabcvr</a>.</span>

	   <span class="site-footer-credits">This page was generated by <a href="https://pages.github.com">GitHub Pages</a> using the <a href="https://github.com/jasonlong/cayman-theme">Cayman theme</a> by <a href="https://twitter.com/jasonlong">Jason Long</a>.</span>
	 </footer>

    </section>


  </body>
  
</html>



