<!DOCTYPE html>
<html lang="en-us">

  <head>
  
    <!-- Global site tag (gtag.js) - Google Analytics -->
	<script async src="https://www.googletagmanager.com/gtag/js?id=UA-127420264-1"></script>
	<script>
	  window.dataLayer = window.dataLayer || [];
	  function gtag(){dataLayer.push(arguments);}
	  gtag('js', new Date());
	  gtag('config', 'UA-127420264-1');
	</script>


    <meta charset="UTF-8">
    <title>CRESS-GLUE by giffordlabcvr</title>
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <link rel="stylesheet" type="text/css" href="../stylesheets/normalize.css" media="screen">
    <link href='https://fonts.googleapis.com/css?family=Open+Sans:400,700' rel='stylesheet' type='text/css'>
    <link rel="stylesheet" type="text/css" href="../stylesheets/stylesheet.css" media="screen">
    <link rel="stylesheet" type="text/css" href="../stylesheets/github-light.css" media="screen">

  </head>

  <body>
    <section class="page-header">
      <h1 class="project-name">CRESS-GLUE</h1>
      <h2 class="project-tagline">A GLUE project for circular Rep-encoding single-stranded DNA (CRESS DNA) viruses.</h2>
      <a href="../index.html" class="btn">Home</a>
      <a href="./project.html" class="btn">Viruses</a>
      <a href="https://github.com/giffordlabcvr/CRESS-GLUE/zipball/master" class="btn">Download .zip</a>
      <a href="https://github.com/giffordlabcvr/CRESS-GLUE/tree/master" class="btn">View on GitHub</a>
    </section>

    <section class="main-content">

	   <h3>
		 <a id="Background" class="anchor" href="#Background" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a><strong>Endogenous CRESS virus (ECV) data</strong>
	   </h3>

		<p>
          We have used GLUE to organise the fossil record of CRESS viruses.
          You can access these data
          <a href="https://github.com/giffordlabcvr/CRESS-GLUE/tree/master">in this repository</a>. 
          This website provides quick links to specific data items. 

		</p>


	   <h3>
		 <a id="ECV reference sequences and data" class="anchor" href="#Ref_ecv_data" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a><strong>ECV reference sequences and data</strong>
	   </h3>

		<p>
           We reconstructed reference sequences for ECVs by aligning ECV sequences that
           derive from the same ancestral germline integration event.
 		</p>
 		
		<p>
          For ECVs derived from <b>circoviruses</b> (genus <i>Circovirus</i>):
         <br>
          <a href="https://github.com/giffordlabcvr/CRESS-GLUE/blob/master/tabular/eve/ecv-circo-refseqs-side-data.tsv"><b>Tabular data</b></a>;
         <br>
          <a href="https://github.com/giffordlabcvr/CRESS-GLUE/tree/master/sources/fasta-refseqs-circo-ecv"><b>Refererence FASTA nucleotide</b></a> sequences (individual FASTA files);
         <br>
          Refererence FASTA protein</b></a> sequences:
          <a href="https://github.com/giffordlabcvr/CRESS-GLUE/blob/master/export/ecv-rep.polypeptide.aln.faa"><b>Rep (aligned)</b></a> (single file);
          <a href="https://github.com/giffordlabcvr/CRESS-GLUE/tree/master/export/ecv-polypeptide-fasta"><b>Rep (unaligned)</b></a> (individual FASTA files);

		</p>
 		
		<p>
          For ECVs derived from <b>cycloviruses</b> (genus <i>Cyclovirus</i>):
         <br>
         <br>
          <a href="https://github.com/giffordlabcvr/CRESS-GLUE/blob/master/tabular/eve/ecv-cyclo-refseqs-side-data.tsv"><b>Tabular data</b></a>.
         <br>
          <a href="https://github.com/giffordlabcvr/CRESS-GLUE/tree/master/sources/fasta-refseqs-cyclo-ecv"><b>Refererence FASTA nucleotide</b></a> sequences (individual FASTA).
         <br>
          <a href="https://github.com/giffordlabcvr/CRESS-GLUE/tree/master/sources/export/"><b>Refererence FASTA protein</b></a> sequences (single file).
          
		</p>


		<p>
          For ECVs derived from <b>CRESS</b> viruses:
         <br>
         <br>
          <a href="https://github.com/giffordlabcvr/CRESS-GLUE/blob/master/tabular/eve/ecv-cress-refseqs-side-data.tsv"><b>Tabular data</b></a>.
         <br>
          <a href="https://github.com/giffordlabcvr/CRESS-GLUE/tree/master/sources/fasta-refseqs-cress-ecv"><b>Refererence FASTA nucleotide</b></a> sequences (individual FASTA).
         <br>
          Refererence FASTA protein</b></a> sequences (single file)
          <a href="https://github.com/giffordlabcvr/CRESS-GLUE/tree/master/sources/export/"><b>Rep (aligned)</b></a>.
          <a href="https://github.com/giffordlabcvr/CRESS-GLUE/tree/master/sources/export/"><b>Rep (aligned)</b></a>.
         
		</p>
		<p>
           Reference sequences were constructed by first constructing alignments of ECV
           sequences that are derived from the same initial germline colonisation event.
           These include orthologous elements in distinct species, and paralogous 
           elements that have arisen via intragenomic duplication of ECV sequences
           (e.g. mediated by segmental duplication or transposition).
  		</p>





	   <h3>
		 <a id="Raw ECV sequences and data" class="anchor" href="#Raw_ecv_data" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a><strong>Raw ECV sequences and data</strong>
	   </h3>

		<p>
          These are the raw data generated by
          <a target="_blank" href="https://giffordlabcvr.github.io/DIGS-tool/">database-integrated genome screening (DIGS)</a>.
          The tabular files contain information about the genomic location of each EVE. 
          EVEs were classified by comparison to a polypeptide sequence reference library
          designed to represent the known diversity of CRESS viruses - this includes extinct 
          lineages represented only by endogenous viral elements (EVEs).
 		</p>

		<p>
          The raw ECV locus data for circoviruses (genus <i>Circovirus</i>)
          and circovirus-like viruses are
          <a href="https://github.com/giffordlabcvr/CRESS-GLUE/blob/master/tabular/eve/ecv-circo-side-data.tsv"><b>here</b></a>.
          These data were obtained via
          <a target="_blank" href="https://giffordlabcvr.github.io/DIGS-tool/">DIGS</a>
          performed in vertebrate genome assemblies downloaded from 
          <a href="https://www.ncbi.nlm.nih.gov/genome/browse ">NCBI genomes</a>
          (2020-07-15).
		</p>

		<p>
          The raw ECV locus data for cycloviruses (genus <i>Cyclovirus</i>) are
          <a href="https://github.com/giffordlabcvr/CRESS-GLUE/blob/master/tabular/eve/ecv-cyclo-side-data.tsv"><b>here</b></a>.
          These data were obtained via
          <a target="_blank" href="https://giffordlabcvr.github.io/DIGS-tool/">DIGS</a>
          performed in invertebrate genome assemblies obtained from 
          <a href="https://www.ncbi.nlm.nih.gov/genome/browse ">NCBI genomes</a>
          (2020-07-15).
 		</p>

		<p>
          The raw ECV locus data for CRESS viruses (genus <i>Cyclovirus</i>) are
          <a href="https://github.com/giffordlabcvr/CRESS-GLUE/blob/master/tabular/eve/ecv-cress-side-data.tsv"><b>here</b></a>.
          These data were obtained via
          <a target="_blank" href="https://giffordlabcvr.github.io/DIGS-tool/">DIGS</a>
          performed in invertebrate genome assemblies obtained from 
          <a href="https://www.ncbi.nlm.nih.gov/genome/browse ">NCBI genomes</a>
          (2020-07-15).
		</p>

		<p> 
          Individual FASTA nucleotide sequences of ECVs derived from cycloviruses are
          <a href="https://github.com/giffordlabcvr/CRESS-GLUE/tree/master/sources/fasta-digs-cyclo-ecv"><b>here</b></a>.
         <br>
         <br>
          Individual FASTA nucleotide sequences of ECVs derived from circoviruses (i.e. genus <i>Circovirus</i>) are
          <a href="https://github.com/giffordlabcvr/CRESS-GLUE/tree/master/sources/fasta-digs-circo-ecv"><b>here</b></a>.
         <br>
         <br>
          Individual FASTA nucleotide sequences of ECVs derived from unclassified CRESS viruses are
          <a href="https://github.com/giffordlabcvr/CRESS-GLUE/tree/master/sources/fasta-digs-cress-ecv"><b>here</b></a>.
         <br>
 		</p>


	   <h3>
		 <a id="Nomenclature" class="anchor" href="#Nomenclature" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a><strong>Nomenclature for ECVs</strong>
	   </h3>

		<p>
          We have applied a systematic approach to naming ECV, following a convention 
          developed for <a href="https://retrovirology.biomedcentral.com/articles/10.1186/s12977-018-0442-1">endogenous retroviruses</a>.
		  Each element was assigned a unique identifier (ID) constructed from a defined
		  set of components.
  		</p>

		<p>
          The first component is the classifier ‘ECV’ (endogenous CRESS virus element). 
		</p>

		<p>
          The second is a composite of two distinct subcomponents separated by a period:
          
          (i)  the name of ECV group;
          (ii) a numeric ID that uniquely identifies the insertion.

		</p>
		
		<p>
          The third component of the ID defines the set of host species in which the ortholog occurs.
          The numeric ID is an integer identifies a unique insertion locus that arose as a
          consequence of an initial germline infection. Thus, orthologous copies in different
          species are given the same number. 
          Where an ECV lineage has been established via intragenomic amplication an
          incorporated ECV sequence, distinct duplicates are identified by numbers after 
          the decimal point.        
 		</p>


	 <footer class="site-footer">
	   <span class="site-footer-owner"><a href="https://github.com/giffordlabcvr/CRESS-GLUE/">CRESS-GLUE</a> is maintained by <a href="https://github.com/giffordlabcvr">giffordlabcvr</a>.</span>

	   <span class="site-footer-credits">This page was generated by <a href="https://pages.github.com">GitHub Pages</a> using the <a href="https://github.com/jasonlong/cayman-theme">Cayman theme</a> by <a href="https://twitter.com/jasonlong">Jason Long</a>.</span>
	 </footer>

    </section>


  </body>
  
</html>



