<!DOCTYPE html>
<html lang="en-us">

  <head>
  
    <!-- Global site tag (gtag.js) - Google Analytics -->
	<script async src="https://www.googletagmanager.com/gtag/js?id=UA-127420264-1"></script>
	<script>
	  window.dataLayer = window.dataLayer || [];
	  function gtag(){dataLayer.push(arguments);}
	  gtag('js', new Date());
	  gtag('config', 'UA-127420264-1');
	</script>


    <meta charset="UTF-8">
    <title>CRESS-GLUE by giffordlabcvr</title>
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <link rel="stylesheet" type="text/css" href="../stylesheets/normalize.css" media="screen">
    <link href='https://fonts.googleapis.com/css?family=Open+Sans:400,700' rel='stylesheet' type='text/css'>
    <link rel="stylesheet" type="text/css" href="../stylesheets/stylesheet.css" media="screen">
    <link rel="stylesheet" type="text/css" href="../stylesheets/github-light.css" media="screen">

  </head>

  <body>
    <section class="page-header">
      <h1 class="project-name">CRESS-GLUE</h1>
      <h2 class="project-tagline">Paleovirology resources for circular Rep-encoding single-stranded DNA (CRESS DNA) viruses.</h2>
      <a href="../index.html" class="btn">Home</a>
      <a href="./project.html" class="btn">Viruses</a>
      <a href="https://github.com/giffordlabcvr/CRESS-GLUE/zipball/master" class="btn">Download .zip</a>
      <a href="https://github.com/giffordlabcvr/CRESS-GLUE/tree/master" class="btn">View on GitHub</a>
    </section>

    <section class="main-content">

	   <h3>
		 <a id="How to use these resources" class="anchor" href="#PaleovirologyUses" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a><strong>How to use these resources</strong>
	   </h3>

		<p>
          We have used GLUE to organise the fossil record of CRESS viruses.
          You can access these data
          <a href="https://github.com/giffordlabcvr/CRESS-GLUE/tree/master">in this repository</a>. 
          This website provides quick links to specific data items. 

		</p>


	   <h3>
		 <a id="ECV reference sequences and data" class="anchor" href="#Ref_ecv_data" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a><strong>ECV reference sequences and data</strong>
	   </h3>

		<p>
           We reconstructed reference sequences for ECVs by aligning ECV sequences that
           derive from the same ancestral germline integration event.
 		</p>
 		


	   <h3>
		 <a id="Multiple sequence alignments and phylogenetic trees" class="anchor" href="#AlignmentsAndTrees" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a><strong>Multiple sequence alignments and phylogenetic trees</strong>
	   </h3>

		<p>
          Multiple sequence alignment constructed in this study are linked together using
          GLUE's 
          ‘<a href="./alignment tree concept small.png">alignment tree</a>’
          data structure. Alignments in the project include:
          <br> 
          <br> (i) A single ‘<a href="https://github.com/giffordlabcvr/CRESS-GLUE/blob/master/alignments/root/cress-root.fna"><strong>root</strong></a>’
          alignment constructed to represent proposed
          homologies between representative members of major CRESS virus lineages
          (including extinct lineages represented only by ECVs).
          <br> (ii) ‘<a href="https://github.com/giffordlabcvr/CRESS-GLUE/tree/master/alignments/tip"><strong>genus-level</strong></a>’
          alignments constructed to represent proposed homologies between the genomes of
          representative members of specific CRESS virus genera and ECV reference sequences.
          <br> (iii) ‘<a href="https://github.com/giffordlabcvr/CRESS-GLUE/tree/master/alignments/tip"><strong>Tip</strong></a>’
          alignments in which all taxa are derived from a single virus species or represent 
          distinct members of a single ECV lineage.
          
 		</p>


	   <h3>
		 <a id="Raw ECV sequences and data" class="anchor" href="#Raw_ecv_data" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a><strong>Raw ECV sequences and data</strong>
	   </h3>

		<p>
          These are the raw data generated by
          <a target="_blank" href="https://giffordlabcvr.github.io/DIGS-tool/">database-integrated genome screening (DIGS)</a>.
          The tabular files contain information about the genomic location of each EVE. 
          EVEs were classified by comparison to a polypeptide sequence reference library
          designed to represent the known diversity of CRESS viruses - this includes extinct 
          lineages represented only by endogenous viral elements (EVEs).
 		</p>






	 <footer class="site-footer">
	   <span class="site-footer-owner"><a href="https://github.com/giffordlabcvr/CRESS-GLUE/">CRESS-GLUE</a> is maintained by <a href="https://github.com/giffordlabcvr">giffordlabcvr</a>.</span>

	   <span class="site-footer-credits">This page was generated by <a href="https://pages.github.com">GitHub Pages</a> using the <a href="https://github.com/jasonlong/cayman-theme">Cayman theme</a> by <a href="https://twitter.com/jasonlong">Jason Long</a>.</span>
	 </footer>

    </section>


  </body>
  
</html>



