<!DOCTYPE html>
<html lang="en-us">

  <head>
  
    <!-- Global site tag (gtag.js) - Google Analytics -->
	<script async src="https://www.googletagmanager.com/gtag/js?id=UA-127420264-1"></script>
	<script>
	  window.dataLayer = window.dataLayer || [];
	  function gtag(){dataLayer.push(arguments);}
	  gtag('js', new Date());
	  gtag('config', 'UA-127420264-1');
	</script>


    <meta charset="UTF-8">
    <title>CRESS-GLUE by giffordlabcvr</title>
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <link rel="stylesheet" type="text/css" href="../stylesheets/normalize.css" media="screen">
    <link href='https://fonts.googleapis.com/css?family=Open+Sans:400,700' rel='stylesheet' type='text/css'>
    <link rel="stylesheet" type="text/css" href="../stylesheets/stylesheet.css" media="screen">
    <link rel="stylesheet" type="text/css" href="../stylesheets/github-light.css" media="screen">

  </head>

  <body>
    <section class="page-header">
      <h1 class="project-name">CRESS-GLUE</h1>
      <h2 class="project-tagline">A GLUE project for CRESS DNA viruses.</h2>
      <a href="../index.html" class="btn">Home</a>
      <a href="./paleovirology.html" class="btn">Paleoviruses</a>
      <a href="https://github.com/giffordlabcvr/CRESS-GLUE/zipball/master" class="btn">Download .zip</a>
      <a href="https://github.com/giffordlabcvr/CRESS-GLUE/tree/master" class="btn">View on GitHub</a>
    </section>

    <section class="main-content">

	   <h3>
		 <a id="Background" class="anchor" href="#Background" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a><strong>Project overview</strong>
	   </h3>

	   <p>
         We have used the
         <a href="http://glue-tools.cvr.gla.ac.uk/"><b>GLUE</b></a>
         software environment to create a sequence data resource
         (‘CRESS-GLUE’) capable of supporting reproducible comparative investigations of
         circular Rep-encoding single-stranded DNA (CRESS DNA) virus genomes.
	   </p>


	   <h3>
		 <a id="sequenceData" class="anchor" href="#sequenceData" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a><strong>CRESS virus sequence data</strong>
	   </h3>

		<p>
		  The sequence data in this project have been organised into the following sources:
		</p>

		<ol>
		  <li><strong>ncbi refseqs</strong>: Genome-length <b>master reference</b> sequences of
		 CRESS DNA virus species representing major lineages, and linked to auxiliary data in		 
		 <a href="https://github.com/giffordlabcvr/CRESS-GLUE/blob/master/tabular/core/cress-refseqs-side-data.tsv">tabular format</a>.
		  <li><strong>ncbi refseqs-circo</strong>: Genome-length reference sequences of
		 representative circovirus species, linked to auxiliary data in		 
		 <a href="https://github.com/giffordlabcvr/CRESS-GLUE/blob/master/tabular/genus/circo-refseqs-side-data.tsv">tabular format</a>.
		 <li><strong>ncbi refseqs-cyclo</strong>: Genome-length reference sequences of
		 representative cyclovirus species, linked to auxiliary data in		 
		 <a href="https://github.com/giffordlabcvr/CRESS-GLUE/blob/master/tabular/genus/cyclo-refseqs-side-data.tsv">tabular format</a>.

		</ol>


		<p>
		  <a href="https://github.com/giffordlabcvr/CRESS-GLUE/tree/master/sources/ncbi-refseqs">XML-formatted files</a>
		  are downloaded directly from <a href="https://www.ncbi.nlm.nih.gov/genbank/">NCBI</a>
		  and are uniquely identified within this project by their <a target="_blank" href="https://www.ncbi.nlm.nih.gov/Sequin/acc.html">GenBank accession numbers</a>.</li> 
		</p>




	   <h3>
		 <a id="Multiple sequence alignments" class="anchor" href="#MultipleSequenceAlignments" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a><strong>Multiple sequence alignments</strong>
	   </h3>

		 <p>
		   Multiple sequence alignment constructed in this study are linked together using
		   GLUE's 
		   ‘<a href="./alignment tree concept small.png">alignment tree</a>’
		   data structure. Alignments in the project include:
		 
		  
		 <ol>
		   <li> A single ‘<a href="https://github.com/giffordlabcvr/CRESS-GLUE/blob/master/alignments/root/cress-root.fna"><strong>root</strong></a>’
		   alignment constructed to represent proposed
		   homologies between representative members of major CRESS virus lineages</li>
		   <li> ‘<a href="https://github.com/giffordlabcvr/CRESS-GLUE/tree/master/alignments/internal/"><strong>genus-level</strong></a>’
		   alignments constructed to represent proposed homologies between the genomes of
		   representative members of specific CRESS virus genera.</li>
		   <li> ‘<a href="https://github.com/giffordlabcvr/CRESS-GLUE/tree/master/alignments/tip"><strong>Tip</strong></a>’
		   alignments in which all taxa are derived from a single virus species.</li>
		 </ol>
		 
		 </p>


	   <h3>
		 <a id="Phylogenetic trees" class="anchor" href="#PhylogeneticTrees" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a><strong>Phylogenetic trees</strong>
	   </h3>

		<p>

          We used GLUE to implement an automated process for deriving midpoint rooted,
          annotated trees from the alignments included in our project.
 		</p>


		<p>
          Trees were constructed at distinct taxonomic levels:
        
		  <ol>
			<li> Recursively populated <a href="https://github.com/giffordlabcvr/CRESS-GLUE/tree/master/trees/main-root"><strong>root</strong></a> phylogeny (Rep)</li>
			<li> <a href="https://github.com/giffordlabcvr/CRESS-GLUE/tree/master/trees/genus-level-with-eves/"><strong>Genus-level</strong></a> phylogenies</li>
			<li> <a href="https://github.com/giffordlabcvr/CRESS-GLUE/tree/master/trees/eve-orthologs/"><strong>ECV lineage-level</strong></a> phylogenies</li>
		  </ol>

 		</p>


	   <h3>
		 <a id="data_quick_links" class="anchor" href="#data_quick_links" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a><strong>Data quick links</strong>
	   </h3>


		<ol>
		  <li>Rep <a href="https://github.com/giffordlabcvr/CRESS-GLUE/blob/master/export/circo-rep.faa">polypeptide sequences</a></li> 
		  <li>Rep <a href="https://github.com/giffordlabcvr/CRESS-GLUE/blob/master/export/circo-rep.fna">nucleotide sequences</a></li> 
		  <li>Cap <a href="https://github.com/giffordlabcvr/CRESS-GLUE/blob/master/export/circo-cap.faa">polypeptide sequences</a></li> 
		  <li>Cap <a href="https://github.com/giffordlabcvr/CRESS-GLUE/blob/master/export/circo-cap.fna">nucleotide sequences</a></li> 
		</ol>


	 <footer class="site-footer">
	   <span class="site-footer-owner"><a href="https://github.com/giffordlabcvr/CRESS-GLUE/">CRESS-GLUE</a> is maintained by <a href="https://github.com/giffordlabcvr">giffordlabcvr</a>.</span>

	   <span class="site-footer-credits">This page was generated by <a href="https://pages.github.com">GitHub Pages</a> using the <a href="https://github.com/jasonlong/cayman-theme">Cayman theme</a> by <a href="https://twitter.com/jasonlong">Jason Long</a>.</span>
	 </footer>

    </section>


  </body>
  
</html>



