<!DOCTYPE html>
<html lang="en-us">

  <head>
  
    <meta charset="UTF-8">
    <title>CRESS-GLUE by giffordlabcvr: An example-based tutorial with parrots</title>
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <link rel="stylesheet" type="text/css" href="../stylesheets/normalize.css" media="screen">
    <link href='https://fonts.googleapis.com/css?family=Open+Sans:400,700' rel='stylesheet' type='text/css'>
    <link rel="stylesheet" type="text/css" href="../stylesheets/stylesheet.css" media="screen">
    <link rel="stylesheet" type="text/css" href="../stylesheets/github-light.css" media="screen">

  </head>

  <body>
    <section class="page-header">
      <h1 class="project-name">CRESS-GLUE</h1>
      <h2 class="project-tagline">A GLUE project for CRESS DNA viruses.</h2>
      <a href="../index.html" class="btn">Home</a>
      <a href="./project.html" class="btn">Viruses</a>
      <a href="./paleovirology.html" class="btn">Paleoviruses</a>
      <a href="https://github.com/giffordlabcvr/CRESS-GLUE/zipball/master" class="btn">Download .zip</a>
      <a href="https://github.com/giffordlabcvr/CRESS-GLUE/tree/master" class="btn">View on GitHub</a>
    </section>

    <section class="main-content">


	   <h3>
		 <a id="sequenceData" class="anchor" href="#sequenceData" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a><strong>How to use CRESS-GLUE - an example-driven tutorial based on an avian circovirus</strong>
	   </h3>


	   <p>
         
         This <b>tutorial</b> comprises the following steps:

		 <ol>
		   <li> <b>Downloading</b> all available sequences of a selected virus from GenBank.</li>
		   <li> <b>Extracting</b> isolate data from GenBank files.</li>
		   <li> <b>Filtering</b> sequences and sequence-associated data to obtain a valid analysis set.</li>
		   <li> <b>Defining virus reference sequences</b> and associating them with annotations.</li>
		   <li> <b>Creating an alignment</b> of all BFDV sequences.</li>
		   <li> <b>Mapping feature coverage</b> in all BFDV isolates.</li>
		   <li> Creating an automated procedure for reconstructing midpoint-rooted <b>phylogenetic trees</b>.</li>
		   <li> Using GLUE projects as a basis for <b>collaborative research and resource development</b>.</li>
		 </ol>


 	   </p>
 	   

	   <p>
         If you haven't decided yet whether or not you want to use GLUE, 
         you might find it useful simply to look through this tutorial, as it gives a 
         good idea of the kind of things that can be done with GLUE projects.
 	   </p>

	   <p>
         If, on the other hand, you've already committed to using GLUE, you 
         may find it useful to investigate the 
         <a target="_blank" href="http://glue-tools.cvr.gla.ac.uk/#/exampleProject"><b>GLUE example project</b></a>
         which provides a broad introduction to the GLUE framework, in addition to
         this tutorial.
 	   </p>


	   <h3>
		 <a id="Background" class="anchor" href="#Background" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a><strong>Background - beak and feather disease virus</strong>
	   </h3>

        <p><img src="./images/parrots.png" alt="Parrots" /></p>


	   <p>
         <b><a target="_blank" href="https://en.wikipedia.org/wiki/Parrot">Psittacine birds</b></a> (parrots and cockatoos) are among the most beautiful and remarkable
         creatures on the planet. Unfortunately, many of these wonderful species are 
         presently endangered. 
         Conservation efforts for have been hampered by
         the emergence of
         <a target="_blank" href="https://www.sciencedirect.com/topics/veterinary-science-and-veterinary-medicine/psittacine-beak-and-feather-disease"><b>psittacine beak and feather disease (PBFD)</b></a>
         in captive populations. PBFD is caused by a <b>CRESS DNA virus</b> belonging to the 
         family <i>Circoviridae</i>, genus <i>Circovirus</i> called <b>beak and feather disease virus (BFDV)</b>.
 	   </p>

	   <p>
         Genomic approaches have great potential to help control the spread of BFDV.
         In this tutorial we will use the <b>CRESS-GLUE</b> project to <b>(i)</b>
         investigate BFDV distribution diversity and evolution, and <b>(ii)</b> establish
         a sequence-based resource for BFDV.
 	   </p>


	   <h3>
		 <a id="DownloadFromNcbi" class="anchor" href="#sequenceData" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a><strong>Downloading all available BFDV sequences from GenBank</strong>
	   </h3>


	   <p>
         To download all BFDV entries in NCBI GenBank, we 
         will use a customised version of GLUE's  module
         <a target="_blank" href="http://glue-tools.cvr.gla.ac.uk/#/moduleReference/moduleType/ncbiImporter">'<b>NcbiImporter</b>'</a>       
         module. Our project-specific configuration of the module can be viewed
		 <a target="_blank" href="https://github.com/giffordlabcvr/CRESS-GLUE/blob/master/modules/genus/circo/circoviridaeNcbiImporterBfdv.xml"><b>here</b></a>.
         Viewing the file, you can probably see that it is configured to download sequences
         based on a query phrase:

	   <p>
        <i>"Beak and feather disease virus"[Organism] AND 200:2000[SLEN]</i>
 	   </p>


	   <p>        
         This 'eSearchTerm' is a standard NCBI entrez text query that
         specifies all GenBank entries labelled "beak and feather disease" in the
         'Organism' field and between 200-2000 nucleotides in length.

 	   </p>

	   <p>
         To use the module, first initiate GLUE on the command line as follows: 
         
         
  <pre>
  <code>MyComputer:CRESS-GLUE rob$ gluetools.sh 
  GLUE Version 1.1.103
  Copyright (C) 2015-2020 The University of Glasgow
  This program comes with ABSOLUTELY NO WARRANTY. This is free software, and you
  are welcome to redistribute it under certain conditions. For details see
  GNU Affero General Public License v3: http://www.gnu.org/licenses/

  Mode path: /
  GLUE></code></pre>


          
         Notice from the first line that I'm in the 'CRESS-GLUE' directory when
         I initiate GLUE.
         This means that, by default, it will be my 'working directory' - when I reference
         a file from the GLUE console, I'll need to do so <u>relative to this directory</u>.
 	   </p>

	   <p>
 
         Next, navigate to the 'cress' project as shown here:
	   </p>

  <pre>
  <code>GLUE> project cress 
  OK
  Mode path: /project/cress/</code></pre>

 	   </p>
 
	   <p>
         Now create the module using it's
		 <a target="_blank" href="https://github.com/giffordlabcvr/CRESS-GLUE/blob/master/modules/genus/circo/circoviridaeNcbiImporterBfdv.xml"><b> configuration file</b></a>,
         which is contained in the CRESS-GLUE project, as shown here:

  <pre>
  <code>Mode path: /project/cress/
  GLUE> create module -f modules/genus/circo/circoviridaeNcbiImporterBfdv.xml
  OK
  (1 Module created) </code></pre>

 	   </p>

 	   <p>
         To run the module, execute the following command in the GLUE shell:

  <pre>
  <code>Mode path: /project/cress/
  GLUE> module circoviridaeNcbiImporterBfdv import</code></pre>

 	   </p>
 	   <p>

         When I ran this command on the 1st December 2020, I obtained the following output:


  <pre>
  <code>ncbiImporterSummaryResult
  totalMatching: 1250
  present: 0
  surplus: 0
  missing: 1250
  deleted: 0
  downloaded: 1250</code></pre>

         Since the GenBank database is continually expanding, you may find
         that you obtain more sequences than this running the same 
         command at a later date.
	   </p>
 
 	   <p>
 	     Next, export the sequences to a 'source' directory as follows.
 	   

   <pre>
   <code>Mode path: /project/cress
   GLUE> export source ncbi-curated-circo-bfdv -p sources/</code></pre>

		  The sequences are now in your local copy of the CRESS-GLUE project, under
		  './sources/ncbi-curated-circo'. However, they are in XML format - what if 
		  you want to get your hands on the FASTA?
		</p>

 	   <p>
          GLUE is designed to provide as much flexibility as possible with respect 
          to data-formatting. To exporting FASTA-formatted sequences from GLUE, use a 
          FASTA exporter module. You can create one with a default configuration on
          the GLUE command line, like this:
          
   <pre>
   <code>Mode path: /project/cress
  GLUE> create module -t fastaExporter 
  OK
  (1 Module created)</code></pre>
		</p>

 	   <p>

          Now lets use this module to export the sequence in the 'ncbi-curated-circo-bfdv'
          source. Heres my first attempt:
   
  <pre>
  <code>Mode path: /project/cress
  GLUE> module fastaExporter export -w "source.name = 'ncbi-curated-circo-bfdv'"
  Error: Usage error
  Usage: export (-w &ltwhereClause> | -a) [-o &ltoffset> -b &ltbatchSize>] [-y &ltlineFeedStyle>]
                [-r] [-t] (-p | -f &ltfileName>)

  Options:
	-y &ltlineFeedStyle>, --lineFeedStyle &ltlineFeedStyle>  LF or CRLF
	-f &ltfileName>, --fileName &ltfileName>                 FASTA file
	-w &ltwhereClause>, --whereClause &ltwhereClause>        Qualify exported sequences
	-o &ltoffset>, --offset &ltoffset>                       Paged query offset
	-b &ltbatchSize>, --batchSize &ltatchSize>               Paged query batch size
	-a, --allSequences                                   Export all project sequences
	-r, --suppressReverseComplement                      Suppress reverse complement
	-t, --suppressRotation                               Suppress rotation
	-p, --preview</code></pre>
  
  	   </p>

 	   <p>
          Oops!! Something went wrong - GLUE needs another argument.
  	   </p>

 	   <p>
          Note that the error message prompts me with correct usage of the module,
          and an explanation of the options. In this case, I forgot to tell GLUE what to do
          with the sequences. They can be previewed in the console, or exported to a file.
          Run the module as follows to export the sequences to a file named 'bfdv.fna':
          
  <pre>
  <code>Mode path: /project/cress
  GLUE> module fastaExporter export -w "source.name = 'ncbi-curated-circo-bfdv'" -f bfdv.fna
  OK</code></pre>
          
   
 	   <h3>
		 <a id="ParseGenBankXML" class="anchor" href="#sequenceData" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a><strong>Importing BFDV sequences and metadata from GenBank files</strong>.
	   </h3>


	   <p>
         Now that we've downloaded a set of BFDV sequences, we can incorporate them into 
         the CRESS-GLUE project.
         Load the sequences as follows:

  <pre>
  <code>Mode path: /project/cress
  GLUE> import source sources/ncbi-curated-circo-bfdv/</code></pre>
 	   </p>
 	   
	   <p>
          All sequences will have the same taxonomic information (since we're only
          considering one virus species here).
          Set the taxonomic data for each sequence in the main 'sequence' table by running
		  <a href="https://github.com/giffordlabcvr/CRESS-GLUE/blob/master/glue/genus/circo/exampleLoadSequenceDataBdfv.js"><b>this script</b></a>
		  as follows:

  <pre>
  <code>Mode path: /project/cress
  GLUE> run script glue/genus/circo/exampleLoadSequenceDataBdfv.js</code></pre>
 	   
	   <p>
          Link each new BFDV sequence to the isolate table using
          <a href="https://github.com/giffordlabcvr/CRESS-GLUE/blob/master/glue/genus/circo/exampleLoadSequenceDataBdfv.js"><b>this script</b></a>.
  <pre>
  <code>Mode path: /project/cress
  GLUE> run script glue/genus/circo/exampleLoadIsolateDataBdfv.js</code></pre>

 	   </p>


	   <p>
         Extracting information from GenBank fields using a
         customised version
         of GLUE's 'genbankPopulator' module.
         We can use this module to search for for <i>host species</i>,
         <i>time and place of isolation</i>,
         and <i>isolate</i> data in the "notes" section of the GenBank XML. 
 	   </p>
 	   
	   <p>
         Use of FreeMarker and REGEX in this and other GLUE modules.
         If unfamiliar the examples included in this project can be modified. 
  
  
  <pre>
  <code>Mode path: /project/cress
  GLUE> run script glue/genus/circo/exampleLoadIsolateDataBdfv.js</code></pre>

 	   </p>


		 <p>

           Inspect data in the sequence table via the command line:

  <pre>
  <code>Mode path: /project/cress
  GLUE> list sequence sequenceID name full_name family genus clade
  +============+===============+================+==================+==============+
  | sequenceID |     name      |     family     |      genus       |    clade     |
  +============+===============+================+==================+==============+
  | AB182568   | BFDV          | Circoviridae   | Circovirus       | Avian-1      |
  | AB182569   | BFDV          | Circoviridae   | Circovirus       | Avian-1      |
  | AB182570   | BFDV          | Circoviridae   | Circovirus       | Avian-1      |
  | AB182571   | BFDV          | Circoviridae   | Circovirus       | Avian-1      |
  | AB182572   | BFDV          | Circoviridae   | Circovirus       | Avian-1      |</code></pre>

  		 </p>


		 <p>
           Inspect data in the isolate table via the command line. This time using a
          <a target="_blank" href="https://www.w3schools.com/sql/sql_where.asp"><b>'where clause'</b></a>         
           to limit the query to Japanese isolates only:

  <pre>
  <code>Mode path: /project/cress
  GLUE> list custom-table-row isolate id isolate country -w "country = 'Japan'"
  +==========+============+=========+
  |    id    |  isolate   | country |
  +==========+============+=========+
  | AB277726 | AGP-SA1P-1 | Japan   |
  | AB277727 | AGP-SA1P-2 | Japan   |
  | AB277728 | AGP-SA1P-3 | Japan   |
  | AB277729 | AGP-SA1P-4 | Japan   |
  | AB277730 | AGP-SA1P-5 | Japan   |</code></pre>		   


  		 </p>

		 <p>
           Executing the query above without a where clause will show that
           not every sequence has complete information for each field in the isolate
           table. This is because typically, some GenBank entries for any given virus
           lack this information.
  		 </p>



	   <h3>
		 <a id="FilterDataSet" class="anchor" href="#sequenceData" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a><strong>Filtering data to obtain a valid analysis data set</strong>.
	   </h3>




		 <p>
           Export data table to a text file (tab-separated values).
           This is a three step process:


  <pre>
  <code>Mode path: /project/cress

  GLUE> console set cmd-output-file-format tab 
  OK
  Mode path: /project/cress
  GLUE> console set next-cmd-output-file bfdv-data.tsv
  OK
  Mode path: /project/cress
  GLUE> list custom-table-row isolate id isolate country host_sci_name collection_year</code></pre>		

		 </p>
		 <p>
           Show format for updating data - (template).
		 </p>

		 <p>
           Use       
		   <a target="_blank" href="https://github.com/giffordlabcvr/CRESS-GLUE/blob/master/glue/genus/circo/exampleImportCuratedSideData.js"><strong>this script</strong></a>
		   to upload the refined data.
		 </p>

		 <p>
		   If you improve metadata for any virus sequence set, you may want to share.
		   See sharing section below.
		 </p>






	   <h3>
		 <a id="RefineReference" class="anchor" href="#sequenceData" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a><strong>Refining the BFDV reference sequence</strong>.
	   </h3>


		 <p>
		   The 'Circovirus' extension of CRESS-GLUE includes reference sequences for 
		   most (if not all) species, including BFDV.
		   The reference sequence declarations can be found
		   <a href="https://github.com/giffordlabcvr/CRESS-GLUE/blob/master/glue/genus/circo/circoReferences.glue"><b>in this file</b></a>.
		 </p>
	   
	   
		 <p>
		   However, in a project focussed specifically on BFDV, we may want to annotate this 
		   specific reference more richly - incorporating everything that is known about
		   the virus genome.
		 </p>

	   
		 <p>
           Let's examine how we can modify the existing BFDV reference sequence.
           First, let's have a look at the reference sequences in the cress project.
           On way to do this is to use the
		   <a href="https://www.howtogeek.com/195207/use-tab-completion-to-type-commands-faster-on-any-operating-system/"><b>tab complete</b></a>           
           function of the console to reveal the full list 
           of reference sequences, by first typing 'reference (or just the first few 
           letters of this word), and then pressing the 'tab' key, as show here:

 <pre>
 <code>Mode path: /project/cress
 GLUE> reference REF_
 REF_Begomo_EACMKV              REF_CRESS1_MASTER_AHEaCV-13    REF_CRESS2_MASTER_Po-41        
 REF_Circo_BFDV                 REF_Circo_BarbCV               REF_Circo_CaCV                 
 REF_Circo_CfCV                 REF_Circo_CoCV                 REF_Circo_DuCV                 
 REF_Circo_FiCV                 REF_Circo_GuCV                 REF_Circo_MASTER_PCV-1         
 REF_Circo_MiCV                 REF_Circo_PCV-2                REF_Circo_PCV-3                
 REF_Circo_RaCV                 REF_Circo_SgCV                 REF_Circo_StCV                 
 REF_Circo_SvCV                 REF_Circo_SwCV                 REF_Circo_TbCV                 
 REF_Circo_ZfCV                 REF_Cyclo_MASTER_HuACyV-1      REF_Porprismaco_MASTER_CBSV    
 REF_Redondo_MASTER_HLVV</code></pre>	        
  	   
  		 </p>
  		 
  		 
		 <p>
           As expected, we see the master references of the core project,
           plus the references associated with the Circovirus extension.
           Navigate to the BFDV reference sequence and list all the features that 
           have been associated with this reference, as follows:

  <pre>
  <code>Mode path: /project/cress
  GLUE> reference REF_Circo_BFDV 
  OK
  Mode path: /project/cress/reference/REF_Circo_BFDV
  GLUE> list feature-location 
  +========================+==============+
  | referenceSequence.name | feature.name |
  +========================+==============+
  | REF_Circo_BFDV         | whole_genome |
  | REF_Circo_BFDV         | Rep          |
  | REF_Circo_BFDV         | Cap          |
  +========================+==============+
  FeatureLocations found: 4</code></pre>	        

  		 </p>
  		 
  		 
		 <p>
 		   The reference has the locations of major genome features mapped - specifically the 
		   <i>rep</i> and <i>cap</i> genes.	   
		   <a href="https://www.ncbi.nlm.nih.gov/nuccore/NC_001944"><b>BFDV GenBank entry</b></a>
		   shows an additional, hypothetical open reading frame (ORF), called "V2".
  		 </p>
  		 
  		 
		 <p>
           Adding V2 modify existing BFDV sequence to include V2. 
           First, lets examine all of the features defined in the entire CRESS-GLUE 
           project.

  <pre>
  <code>GLUE> project-mode 
  OK
  Mode path: /project/cress
  GLUE> list feature
  feature             feature-location    
  GLUE> list feature
  +==============+==============+===================================+
  |     name     | parent.name  |            description            |
  +==============+==============+===================================+
  | whole_genome | -            | Whole genome                      |
  | 5flank       | -            | 5 prime genomic flanking sequence |
  | 3flank       | -            | 3 prime genomic flanking sequence |
  | Rep          | whole_genome | Replicase protein                 |
  | ORFV2        | whole_genome | Hypothetical protein ORFV2        |
  | ORFC2        | whole_genome | Hypothetical protein ORFC2        |
  | ORF4         | whole_genome | Hypothetical protein ORF4         |
  | ORF3         | whole_genome | Hypothetical protein ORF3         |
  | nonanuc      | whole_genome | nonanucleotide motif              |
  | intergenic   | whole_genome | intergenic region                 |
  | gp3          | whole_genome | gp3 protein                       |
  | gp2          | whole_genome | gp2 protein                       |
  | Cap          | whole_genome | Capsid protein                    |
  | 5UTR         | whole_genome | 5' UTR                            |
  | 3UTR         | whole_genome | 5' 3TR                            |
  +==============+==============+===================================+
  Features found: 15</code></pre>


  		 </p>

  		 
		 <p>

  <pre>
  <code>Mode path: /project/cress
  GLUE> create feature v2 "v2 gene"
  OK
  (1 Feature created)</code></pre>



  <pre>
  <code>Mode path: /project/cress
  GLUE> reference REF_Circo_BFDV 
  OK
  Mode path: /project/cress/reference/REF_Circo_BFDV</code></pre>	
  		 </p>



	   <h3>
		 <a id="CreateAlignment" class="anchor" href="#sequenceData" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a><strong>Creating an alignment of all BFDV sequences</strong>.
	   </h3>

	   
		 <p>
		   An efficient and reliable approach for 
		   aligning sequences derived from a single virus species
		   is to use a pairwise, reference guided approach initially. The GLUE
		   framework allows this approach to be used, in a codon-aware way, and
		    without data loss.
		 </p>
	   
		 <p>
		   First, exit the reference sequence to project level (assuming you are still 
		   in this mode path):
		 </p>

   <pre>
   <code>Mode path: /project/cress/reference/REF_Circo_BFDV
   GLUE> project-mode 
   OK
   Mode path: /project/cress
   GLUE></code></pre>
	   
		 <p>
		   Define the alignment.

  <pre>
  <code>Mode path: /project/cress
  GLUE> create alignment AL_BFDV -r REF_Circo_BFDV 
  alignment AL_BFDV add member --whereClause "source.name = 'ncbi-curated-circo-bfdv'"</code></pre>
		 </p>


		 <p>
		   Create the alignment.
		 </p>


		 <p>
		   The CRESS-GLUE project contains a 
		   <a href="https://github.com/giffordlabcvr/CRESS-GLUE/blob/master/modules/genus/circo/circovirusCompoundAligner.xml"><b>circovirus-specific configuration file</b></a>
		   for GLUE's  'compoundAligner' module.
  <pre>
  <code>compute alignment AL_BFDV circovirusCompoundAligner</code></pre>
		 </p>

  

		 <p>
		   Inspect the alignment.

  <pre>
  <code>GLUE> alignment AL_BFDV show statistics</code></pre>

		 </p>

		 <p>
		   Or alternatively, list alignment members, along with their associated data, like this:
		 </p>


  <pre>
  <code>GLUE> alignment AL_BFDV 
  OK
  Mode path: /project/cress/alignment/AL_BFDV
  GLUE> list member sequence.sequenceID sequence.name sequence.isolate.isolate
  +=====================+===============+====================================+
  | sequence.sequenceID | sequence.name |      sequence.isolate.isolate      |
  +=====================+===============+====================================+
  | AB182568            | BFDV          | YCC-IN1P                           |
  | AB182569            | BFDV          | YCC-IN2P                           |
  | AB182570            | BFDV          | RRP-JP11P                          |
  | AB182571            | BFDV          | RRP-JP12P                          |
  | AB182572            | BFDV          | BTP-SA11P                          |</code></pre>
  
     
		 <p>
		   (only the first few results are shown here)
		 </p>

		 <p>

		   Some BFDV sequences were in the opposite orientation than the reference.
		   First, run 
		   <a href="https://github.com/giffordlabcvr/CRESS-GLUE/blob/master/glue/genus/circo/exampleApplyReverseComplement.js"><b>this GLUE file </b></a>
		   to fix this - then I'll explain how I created the file.
		 </p>



  <pre>
  <code>run file glue/genus/circo/exampleApplyReverseComplement.glue</code></pre>


	   
		 <p>
		   How did I figure out which ones? There are lots of ways to do this 
		   using GLUE. In this case I used a fairly simple approach, but one that seemed
		   to work well. I simply aligned all BFDV sequences then expect the alignment.
		 </p>

	   
		 <p>
		   Refining the alignment - may want to manually adjust and re-import.
		   Note, the structure of the alignments directory - refer to alignment tree.
		   BFDV is a 'tip' alignment'.
		 </p>

   <pre>
   <code>Mode path: /project/cress/
   GLUE> import alignment -f alignments/tip/</code></pre>
	   
		 <p>
		   There is in reality only one correct alignment.
		   However, in some highly-variable regions of viral genomes, the concept 
		   might not even be that meaningful.
		 </p>

		 <p>
		   If you improve an alignment in a way that seems clear - you may want to share.
		   See sharing section below.
		 </p>

		 <h3>
		   <a id="MapCoverage" class="anchor" href="#sequenceData" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a><strong>Mapping feature coverage in BFDV sequences</strong>.
		 </h3>

	   
		 <p>
		   Now that we have created an alignment of BFDV sequences that is constrained to 
		   the BFDV reference sequences, we can use this alignment, and features of the 
		   GLUE framework, to examine feature coverage within individual BFDV sequences.
		 </p>

  <pre>
  <code>module circoFeaturePresenceRecorder 
  GLUE> record feature-presence AL_BFDV -r -f whole_genome -d</code></pre>


	   <h3>
		 <a id="buildPhylogenies" class="anchor" href="#sequenceData" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a><strong>Reconstructing phylogenetic trees</strong>.
	   </h3>
	   
	   <p>
         In the final step in this tutorial, we will create a set of midpoint-rooted
         tree for each of the two main genes in the BFDV genome. 
		   <a href="https://github.com/giffordlabcvr/CRESS-GLUE/blob/master/glue/genus/circo/exampleCreateBfdvPhylogenies.glue"><b>this file</b></a>.

  <pre>
  <code>GLUE> alignment AL_BFDV 
  OK
  Mode path: /project/cress/alignment/AL_BFDV
  GLUE> list member sequence.sequenceID sequence.name sequence.isolate.isolate
  +=====================+===============+====================================+
  | sequence.sequenceID | sequence.name |      sequence.isolate.isolate      |
  +=====================+===============+====================================+
  | AB182568            | BFDV          | YCC-IN1P                           |
  | AB182569            | BFDV          | YCC-IN2P                           |
  | AB182570            | BFDV          | RRP-JP11P                          |
  | AB182571            | BFDV          | RRP-JP12P                          |
  | AB182572            | BFDV          | BTP-SA11P                          |</code></pre>
  



        </p>
       
       
		 <p>
		   The finished trees with annotations can be found 
		   <a target="_blank" href="https://github.com/giffordlabcvr/CRESS-GLUE/tree/master/trees/virus-level"><b>here</b></a>.
		 </p>



	   <h3>
		 <a id="CollaborativeUse" class="anchor" href="#sequenceData" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a><strong>Collaboratively developing CRESS-GLUE resources</strong>.
	   </h3>

		 <p>
		   The 'philosophy' of GLUE - open source software for dealing with the
		   nuts and bolts of comparative sequence analysis.
		   The aim of this project really is to <b>facilitate</b>.
		   Comparative analysis hampered by laborious data preparation.
		   
		 </p>
	   
		 <p>
		   Host projects on GitHub.
		   Clone. fork.  
		 </p>



	   <h3>
		 <a id="Publications" class="anchor" href="#Publications" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a><strong>Related Publications</strong>
	   </h3>

        Arvind Varsani, Regnard GL, Bragg R, Hitzeroth II, EP Rybicki
		<b>(2011)</b>
		<br>
		Global genetic diversity and geographical and host-species distribution of beak and feather disease virus isolates.
		<i>J Gen Virol.</i>  Apr;92(Pt 4):752-67. doi: 10.1099/vir.0.028126-0. Epub 2010 Dec 22. 
		<a href="https://www.microbiologyresearch.org/content/journal/jgv/10.1099/vir.0.028126-0" target="_blank">[view]</a> 
		<br>

        <p><img src="./images/parrots.png" alt="Parrots" /></p>

	   <h3>
		 <a id="Related resources" class="anchor" href="#glueProject" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a><strong>Related resources</strong>
	   </h3>
 
		 <p>
		   Parrot conservation societies:
		   FreeMarker
		   GLUE example project
		 </p>


	  <footer class="site-footer">
	 
		<span class="site-footer-owner"><a href="https://github.com/giffordlabcvr/CRESS-GLUE/">
		  CRESS-GLUE</a> is maintained by <a href="https://github.com/giffordlabcvr">giffordlabcvr</a>.
		</span>

		<span class="site-footer-credits">
		  This page was generated by <a href="https://pages.github.com">GitHub Pages</a> using the <a href="https://github.com/jasonlong/cayman-theme">Cayman theme</a> by <a href="https://twitter.com/jasonlong">Jason Long</a>.
		</span>
	   
	  </footer>

    </section>


  </body>
  
</html>



